software/contrib/pams.py

```diff
@@ -0,0 +1,1126 @@
+#!/usr/bin/env python3
+"""A EuroPi clone of ALM's Pamela's NEW Workout
+
+@author Chris Iverach-Brereton <ve4cib@gmail.com>
+@year   2023
+
+See pams.md for complete feature list
+"""
+
+from europi import *
+from europi_script import EuroPiScript
+
+from contrib.euclid import generate_euclidean_pattern
+from contrib.quantizer import Quantizer
+from contrib.screensaver import Screensaver
+
+from machine import Timer
+
+import math
+import time
+import random
+
+## The version of Pam's Workout
+#
+#  Ideally this should sync 1:1 with a tag in the development branch
```

Comment:
This is mostly a leftover from the generall call for feedback; I figured I couldn't rely on people necessarily testing with my latest `HEAD`, so I figured an internal version number would help with that.  I can remove it though, since I don't think it's necessary to have specific versions for individual scripts.